beacon-chain/operations/attestations/aggregate.go

```diff
@@ -0,0 +1,77 @@
+package attestations
+
+import (
+	"context"
+	"time"
+
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/go-ssz"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"go.opencensus.io/trace"
+)
+
+// Define time to aggregate the unaggregated attestations at 3 times per slot, this gives
+// enough confidence all the unaggregated attestations will be aggregated as aggregator requests.
+var timeToAggregate = time.Duration(params.BeaconConfig().SecondsPerSlot/3) * time.Second
+
+// This kicks off a routine to aggregate the unaggregated attestations from pool.
+func (s *Service) aggregateRoutine() {
+	ticker := time.NewTicker(timeToAggregate)
+	ctx := context.TODO()
+	for {
+		select {
+		case <-s.ctx.Done():
+			return
+		case <-ticker.C:
+			unaggregatedAtts := s.pool.UnaggregatedAttestations()
+			if err := s.aggregateAttestations(ctx, unaggregatedAtts); err != nil {
+				log.WithError(err).Error("Could not aggregate attestation")
+			}
+		}
+	}
+}
+
+// This aggregates the input attestations via AggregateAttestations helper
+// function.
+func (s *Service) aggregateAttestations(ctx context.Context, unaggregatedAtts []*ethpb.Attestation) error {
+	_, span := trace.StartSpan(ctx, "Operations.attestations.aggregateAttestations")
```

Comment:
```suggestion
	ctx, span := trace.StartSpan(ctx, "Operations.attestations.aggregateAttestations")
```

Please avoid this pattern, it can lead to using the wrong context in the future.

Reply:
Ah thanks. I'll fix it for rest of the other places too